libexec/rtld-elf/rtld.c

```diff
@@ -1467,6 +1522,13 @@ digest_dynamic1(Obj_Entry *obj, int early, const Elf_Dyn **dyn_rpath,
 	    assert(dynp->d_un.d_val == sizeof(Elf_Sym));
 	    break;
 
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+	case DT_CHERI_C18N_SIG:
```

Comment:
What is the best way to allow people to use it without needing to patch rtld themselves? Could I put this behind a env var like `LD_C18N_EXPERIMENTAL`?

Reply:
You could, but it will be code that sees zero testing in CI. I'd personally rather see it evolve and mature before being included in a release. But if it requires explicit opt-in and thus won't be latent code that suddenly starts doing things once a toolchain emits entries with this tag (which may not in future be compatible with what this code does today), if it even ends up using this tag as the encoding (which is questionable, given you've put it in the OS space but it has a very CPU-specific meaning), then I won't *object* to it. Perhaps one to ask @gnn about.